agenthub/codeact_agent/codeact_agent.py

```diff
@@ -243,30 +240,33 @@ def _get_messages(self, state: State) -> list[Message]:
             if message:
                 # handle error if the message is the SAME role as the previous message
                 # litellm.exceptions.BadRequestError: litellm.BadRequestError: OpenAIException - Error code: 400 - {'detail': 'Only supports u/a/u/a/u...'}
-                # there should not have two consecutive messages from the same role
+                # there shouldn't be two consecutive messages from the same role
                 if messages and messages[-1].role == message.role:
                     messages[-1].content.extend(message.content)
                 else:
                     messages.append(message)
 
         # Add caching to the last 2 user messages
         if self.llm.supports_prompt_caching:
-            user_turns_processed = 0
-            for message in reversed(messages):
-                if message.role == 'user' and user_turns_processed < 2:
-                    message.content[
-                        -1
-                    ].cache_prompt = True  # Last item inside the message content
-                    user_turns_processed += 1
-
-        # the latest user message is important:
+            user_messages = list(
+                islice((m for m in reversed(messages) if m.role == 'user'), 2)
+            )
```

Comment:
How about more explicit slicing for better readability?
```suggestion
            user_messages = list((m for m in reversed(messages) if m.role == 'user'))[2:]
```

Reply:
I think the "islice" performs better if I understood the description correctly: it should actually stop iterating the list when 2 are reached, whereas in your suggested code, ALL messages are iterated first and then sliced? 🤔